### PR TITLE
Allow yaml in script and automation

### PR DESCRIPTION
**Pull request by bramkragten**
Part of https://github.com/home-assistant/home-assistant-polymer/issues/3784

### CODE REVIEW

File: src/panels/config/js/yaml_textarea.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+export default class YAMLTextArea extends Component<any, any> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isvalid: true,
+      value:
+        props.value && !isEmpty(props.value)
+          ? yaml.safeDump(props.value)
```
Comment (balloob) [inline src/panels/config/js/yaml_textarea.tsx]: What if this raises ?

Comment (balloob) [reply src/panels/config/js/yaml_textarea.tsx]: `yaml.safeDump(() => {})`

**Review by balloob — APPROVED** [review]

File: src/panels/config/js/yaml_textarea.tsx [hunk 2]
```diff
@@ -14,12 +14,20 @@ const isEmpty = (obj) => {
...
+
+    let value: string | undefined;
+    try {
+      value =
         props.value && !isEmpty(props.value)
           ? yaml.safeDump(props.value)
-          : undefined,
+          : undefined;
+    } catch (err) {
+      alert(`There is an error in the YAML: ${err}`);
```
Comment (balloob) [inline src/panels/config/js/yaml_textarea.tsx]: Not an error in the YAML, could not convert it to YAML .

Comment (bramkragten) [reply src/panels/config/js/yaml_textarea.tsx]: But we don't have any other way to save the config than YAML? So where could the error come from? YAML -> JSON -> YAML?

Comment (balloob) [reply src/panels/config/js/yaml_textarea.tsx]: Good question, I don't know :D

Comment (balloob) [reply src/panels/config/js/yaml_textarea.tsx]: It's fine.

**Pull request merged**